docs/usage/configuration-options.md

```diff
@@ -2039,6 +2039,10 @@ In case there is a need to configure them manually, it can be done using this `r
 
 The field supports multiple URLs however it is datasource-dependent on whether only the first is used or multiple.
 
+## replacementName
+
+## replacementVersion
+
```

Comment:
Needs expanding, including if both are mandatory or only `replacementName`.

Reply:
Fair point. Other changes indicate that the both of them are required. Allowing the both of them to be optional would allow for easier integration with existing available metadata, like [Maven's relocation](https://maven.apache.org/guides/mini/guide-relocation.html) information. With that, the `groupId`, `artifactId` and `version` can all be `null`.

Reply:
Would it make sense to add these fields to the `validMatchFields` of [the regex manager](lib/manager/regex/index.ts)? I already have scenarios where, based on certain regex matches/capture groups, I wanna compose a replacement version string.

Reply:
@rarkins Can do

@astellingwerf Yes, but I would like to get some form of this merged in a _very_ limited state, and then build on top of that. Adding support for the regex manager should come in a later iteration, in my opinion. This is already a large, long-lived, pull request—though that is, admittedly, my fault.

Reply:
Would you then recommend me to hold off on my MR, and create a separate issue and MR to add replacement support for Gradle/Maven/Java/...?

Reply:
@astellingwerf Sorry, I missed your response earlier. I agree with holding off for now, and creating followup issues/PRs